src/stream/src/executor/mod.rs

```diff
@@ -297,19 +297,23 @@ pub enum Mutation {
 }
 
 #[derive(Debug, Clone)]
-pub struct Barrier {
+pub struct BarrierInner<M> {
```

Comment:
Can we add some docs specifying the candidates of `M` and the corresponding cases?